src/transformers/generation/utils.py

```diff
@@ -1557,6 +1582,16 @@ def generate(
                 )
             if generation_config.cache_implementation == "static":
                 model_kwargs["past_key_values"] = self._get_static_cache(batch_size, generation_config.max_length)
+            elif generation_config.cache_implementation == "sliding_window":
```

Comment:
maybe the check for sliding window can go in the init of the cache class instead of here

Reply:
I have put the logic inside `_get_cache`

Reply:
Yeah, I have moved them inside init